llvm/test/SYCLLowerIR/CompileTimePropertiesPass/fp-control-kernel-properties.ll

```diff
@@ -0,0 +1,182 @@
+; RUN: opt -passes=compile-time-properties %s -S | FileCheck %s
+
+
+define spir_kernel void @"Kernel0"() #0 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel1"() #1 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel2"() #2 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel3"() #3 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel4"() #4 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel5"() #5 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel6"() #6 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel7"() #7 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel8"() #8 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel9"() #9 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel10"() #10 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel11"() #11 {
+entry:
+  ret void
+}
+
+define spir_kernel void @"Kernel12"() #12 {
+entry:
+  ret void
+}
+
+; SPIRV execution modes for FP control.   | BitMask
+; ROUNDING_MODE_RTE = 4462;               | 00000000000
+; ROUNDING_MODE_RTP_INTEL = 5620;         | 00000010000
+; ROUNDING_MODE_RTN_INTEL = 5621;         | 00000100000
+; ROUNDING_MODE_RTZ = 4463;               | 00000110000
+; DEMORM_FLUSH_TO_ZERO = 4460;            | 00000000000
+; DENORM_PRESERVE (double) = 4459;        | 00001000000
+; DENORM_PRESERVE (float) = 4459;         | 00010000000
+; DENORM_PRESERVE (half) = 4459;          | 10000000000
+
+; rte + ftz (Default)
+; CHECK: !0 = !{ptr @Kernel0, i32 4462, i32 64}
```

Comment:
I'm a bit concerned about amount of repeated opcodes in here, i.e. it seems like there is tons of places to make a typo.

One option to improve is to use [FileCheck String Substitution Blocks](https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-string-substitution-blocks) to capture opcode once and then re-use it. To capture it uses regexp, but I think it should also work if you spell it like: `[[RTE:4462]]` to ensure that we have captured the right opcode in the first place.

Alternative approach is to use [`-DVAR`](https://llvm.org/docs/CommandGuide/FileCheck.html#cmdoption-FileCheck-D-VAR) arguments to explicitly set variables with those opcodes through command line and check for them instead of using hardcoded numbers all over the place.

My concern is minor, so I'm ok to leave it as-is. Would be good to hear feedback from other reviewers as well

Reply:
Thank you for the suggestion, fixed to use Substitution blocks.